### DISCUSSION TITLE
Use case code error: KeyError: 'KRATOS_ ROOT_ PATH'

### DISCUSSION THREAD

**Discussion by hhcyd** [KratosMultiphysics/Kratos#10828] [Q&A] [answered]
I downloaded someone else's case online and ran the python code with kratos7.0 version, which showed that the error occurred in this place: kratos_ root_ path = os.environ['KRATOS_ROOT_PATH']。 The error prompt is KeyError: 'KRATOS_ ROOT_ PATH'

Based on the error reported by python code, I guess there is no KRATOS_ ROOT_ PATH is the file path or environment. In addition, there are no StructuralApplication, EkateAuxiliaryApplication, MKLSolversApplication and MortarApplication. Could you tell me whether these applications are lost due to the kratos version update or whether the environment and programs need to be reconfigured before the code runs. What should I do to make this code run smoothly?
![image](https://user-images.githubusercontent.com/120029178/222982089-2ea6d565-f332-47b2-b157-8d4ac74bad42.png)
![image](https://user-images.githubusercontent.com/120029178/222982181-cd0744af-dadc-4581-bdef-ae3fbd933aca.png)

**Comment by hhcyd** [comment]
This is a file in the txt format of python code. Please see if there are any problems with the code.
[test.txt](https://github.com/KratosMultiphysics/Kratos/files/10892794/test.txt)

**Comment by roigcarlo** [comment, marked as answer]
Dear @hhcyd 

I am not familiar at all with this code. 

`KRATOS_ ROOT_ PATH` is not related with any particular Kratos feature. `os.environ` references variables in your developing environment, so that must be something specific the person that created the script is using. 

Also I don't recognize most of the applications used in the script (others are just outdated). My guess is that you are using some kind of extension that someone has developed, so I would suggest to ask first the person that developed this particular script.

Please don't hesitate to check any of our examples in the repository:
https://github.com/KratosMultiphysics/Examples

that will provide you with a more up to date version of a standard Kratos script.
